src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -168,9 +169,14 @@ def serialize(self, parent_map, root, warning_extra, params, snuba_event=None, e
                 event.event_id: event
                 for event in eventstore.get_events(
                     eventstore.Filter(
+                        start=params["start"],
+                        end=params["end"],
                         project_ids=params["project_id"],
-                        event_ids=[event["id"] for event in parent_map.values()],
-                    )
+                    ),
+                    event_list=[
+                        Event(project_id=event["project.id"], event_id=event["id"])
+                        for event in parent_map.values()
```

Comment:
The eventstore should create the actual event instances, like it does in the rest of this method. Doing it this way only if an event list is specified is rather inconsistent. I think it'd be better to just pass a list of project id / event id tuples.